octobot_trading/exchanges/implementations/spot_ccxt_exchange.py

```diff
@@ -161,23 +211,34 @@ async def _create_market_sell_order(self, symbol, quantity, price=None, params=N
     async def _create_limit_sell_order(self, symbol, quantity, price=None, params=None) -> dict:
         return await self.connector.client.create_limit_sell_order(symbol, quantity, price, params=params)
 
-    async def _create_market_stop_loss_order(self, symbol, quantity, price=None, params=None) -> dict:
-        return None
+    async def _create_market_stop_loss_order(self, symbol, quantity, price, side, current_price, params=None) -> dict:
+        raise NotImplementedError("_create_market_stop_loss_order is not implemented")
 
-    async def _create_limit_stop_loss_order(self, symbol, quantity, price=None, params=None) -> dict:
-        return None
+    async def _create_limit_stop_loss_order(self, symbol, quantity, price=None, side=None, params=None) -> dict:
+        raise NotImplementedError("_create_limit_stop_loss_order is not implemented")
 
-    async def _create_market_take_profit_order(self, symbol, quantity, price=None, params=None) -> dict:
-        return None
+    async def _create_market_take_profit_order(self, symbol, quantity, price=None, side=None, params=None) -> dict:
+        raise NotImplementedError("_create_market_take_profit_order is not implemented")
 
-    async def _create_limit_take_profit_order(self, symbol, quantity, price=None, params=None) -> dict:
-        return None
+    async def _create_limit_take_profit_order(self, symbol, quantity, price=None, side=None, params=None) -> dict:
+        raise NotImplementedError("_create_limit_take_profit_order is not implemented")
 
-    async def _create_market_trailing_stop_order(self, symbol, quantity, price=None, params=None) -> dict:
-        return None
+    async def _create_market_trailing_stop_order(self, symbol, quantity, price=None, side=None, params=None) -> dict:
+        raise NotImplementedError("_create_market_trailing_stop_order is not implemented")
 
-    async def _create_limit_trailing_stop_order(self, symbol, quantity, price=None, params=None) -> dict:
-        return None
+    async def _create_limit_trailing_stop_order(self, symbol, quantity, price=None, side=None, params=None) -> dict:
+        raise NotImplementedError("_create_limit_trailing_stop_order is not implemented")
+
+    def _get_ccxt_order_type(self, order_type: enums.TraderOrderType):
```

Comment:
Should it also be implemented in `future_ccxt_exchange`?

Reply:
I think we should put this into `CCXTExchange` directly, it's the same in future and spot